### ISSUE TITLE
esp8266 WeMos D1 Mini. Blinky is compiled and  flashed but does not work. 

### ISSUE THREAD

**Issue by Evgsurf** [pfalcon/esp-open-sdk#330]
Hello,

I try to use  esp-open-sdk with esp8266 WeMos D1 Mini.

I build esp-open-sdk under Ubuntu 18/04  64. 
Everything is ok.

I connect esp to PC via USB cable and get /dev/ttyUSB0 device (WeMos has USB/Serail on board).

Well, let try to compile blinky and flash it.
```


~/esp-open-sdk/examples/blinky$ esptool version
esptool.py v2.1
2.1

~/esp-open-sdk/examples/blinky$ esptool flash_id
Detecting chip type... ESP8266
Chip is ESP8266
Enabling default SPI flash mode...
Manufacturer: c8
Device: 4016
Detected flash size: 4MB


~/esp-open-sdk/examples/blinky$ make
xtensa-lx106-elf-gcc -Teagle.app.v6.ld  blinky.o  -nostdlib -Wl,--start-group -lmain -lnet80211 -lwpa -llwip -lpp -lphy -lc -Wl,--end-group -lgcc -o blinky
esptool.py elf2image blinky
esptool.py v1.2


~/esp-open-sdk/examples/blinky$ make flash
esptool.py write_flash 0 blinky-0x00000.bin 0x10000 blinky-0x10000.bin
esptool.py v1.2
Connecting...
Auto-detected Flash size: 32m
Running Cesanta flasher stub...
Flash params set to 0x0040
Writing 36864 @ 0x0... 36864 (100 %)
Wrote 36864 bytes at 0x0 in 3.2 seconds (92.3 kbit/s)...
Writing 200704 @ 0x10000... 200704 (100 %)
Wrote 200704 bytes at 0x10000 in 17.4 seconds (92.5 kbit/s)...
Leaving...


~/esp-open-sdk/examples/blinky$ miniterm /dev/ttyUSB0  74800
--- Miniterm on /dev/ttyUSB0  74800,8,N,1 ---
--- Quit: Ctrl+] | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H ---

 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x40100000, len 31364, room 16 
tail 4
chksum 0xef
load 0x00000000, len 0, room 4 
tail 0
chksum 0xef
ho 12 tail 0 room 4
load 0x00000000, len 0, room 12 
tail 0
chksum 0xef
csum 0xef
csum err
ets_main.c

```

As you see, blinky was compiled sussessfull, but does not work ... :(

**Comment by Evgsurf** [comment]
BTW hardware is Ok, because I can compile and flash blink in Arduino IDE and it does work.

**Comment by martinberlin** [comment]
Happens the same here. Blinky example just ends up blinking too fast and I get the same error. 
The funny thing is that I had it working before. I've tried with both nodeMCU esp-12 and Wemos D1, it's the same. As Evgsurf compiling something with Arduino goes fine, so it must be some bad setup in ESP Sdk.

**Comment by anthcp** [comment]
I agree, I have the same problem with a NodeMCU esp8266... Audrino for ESP8266 and MicroPython work fine but this SDK seems to be creating binaries that do not run properly...

**Comment by Evgsurf** [comment]
Hi guys. 
I  tried so many things ... no result. I can flash but hardware is crushed.
Of couse I can flash and start hardware via Arduino IDE.
I tired take a look Arduino command. It is really long and has huge characters ...
As I understand now, the problem is old Wemos D1 Mini card.
![p80830-152322](https://user-images.githubusercontent.com/853852/44851675-c2877180-ac69-11e8-9852-88f366a2988b.jpg)
![p80830-152311](https://user-images.githubusercontent.com/853852/44851676-c2877180-ac69-11e8-86f1-679dd070bb9c.jpg)

I tries with another "Wemos D1 Mini" (quoted because it is not really wemos) and can flash and start hardware without any problems.
![p80830-152255 1](https://user-images.githubusercontent.com/853852/44851767-0b3f2a80-ac6a-11e8-9cab-b02e16f60766.jpg)
![p80830-152317](https://user-images.githubusercontent.com/853852/44851768-0b3f2a80-ac6a-11e8-8590-683b1b055436.jpg)


I tried NodeMCU, and it works well too
![p80830-152328](https://user-images.githubusercontent.com/853852/44851793-1eea9100-ac6a-11e8-90a7-bde615a53d80.jpg)

**Comment by martinberlin** [comment]
Note: This is because you are not sending the ESP init data. 
Read this: https://github.com/nodemcu/nodemcu-firmware/blob/master/docs/en/flash.md

You should burn also in the right esp_init data in the right location for each of this SoC. Very nice pictures of the chips, but this Issues, are more meant for Technical issues ;)

**Comment by NicosKaralis** [comment]
Did someone solve this issue? I'm still having this problem

@martinberlin Can you provide a new link? The one you provided no longer works

**Comment by martinberlin** [comment]
https://nodemcu.readthedocs.io/en/master/flash/
Read the part: SDK init Data
